server/src/modules/chatRooms/controllers/createChatRoom.ts

```diff
@@ -7,51 +7,64 @@ import { AuthenticatedUserDataRequest } from '../../../interfaces';
+  if (chats.data.chat.length>0 ){
+    res.status(500).json({ message: 'Já existe um chat room com este feed de origem' });
+    return;
+  }
+
   const feedOrigin = await getFeedMessage(messageId);
 
-  const owner = (req as AuthenticatedUserDataRequest).userId;
+  if (!feedOrigin.error){
 
-  if (feedOrigin.data.feedMessage[0].owner != owner) {
-    res.status(500).json({ message: 'usuário não é dono do feed de origem' });
-    return;
-  }
+    const owner = (req as AuthenticatedUserDataRequest).userId;
+
+    if (feedOrigin.data.feedMessage[0].owner != owner) {
+      res.status(500).json({ message: 'usuário não é dono do feed de origem' });
+      return;
+    }
+
+    const newChatRoom = {
+      feedMessageOrigin: feedOrigin.data.feedMessage[0]._id,
+      playerCharacters: feedOrigin.data.feedMessage[0].playerCharacters,
+      owner: feedOrigin.data.feedMessage[0].owner,
+      title: feedOrigin.data.feedMessage[0].title,
+      content: feedOrigin.data.feedMessage[0].content,
+      image: feedOrigin.data.feedMessage[0].image,
+      numberOfPlayers: feedOrigin.data.feedMessage[0].numberOfPlayers,
+      waitingForResponse: false,
+      createdAt: feedOrigin.data.feedMessage[0].createdAt,
+    };
+
+    const result = await ChatRoomServices.create(newChatRoom);
 
-  const newChatRoom = {
-    feedMessageOrigin: feedOrigin.data.feedMessage[0]._id,
-    playerCharacters: feedOrigin.data.feedMessage[0].playerCharacters,
-    owner: feedOrigin.data.feedMessage[0].owner,
-    title: feedOrigin.data.feedMessage[0].title,
-    content: feedOrigin.data.feedMessage[0].content,
-    image: feedOrigin.data.feedMessage[0].image,
-    numberOfPlayers: feedOrigin.data.feedMessage[0].numberOfPlayers,
-    waitingForResponse: false,
-    createdAt: feedOrigin.data.feedMessage[0].createdAt,
-  };
-
-  const result = await ChatRoomServices.create(newChatRoom);
-
-  if (!result.error) {
-    // join-chatroom
-    const user = result.data.newChatRoom.owner.toString();
-    const roomId = result.data.newChatRoom._id.toString();
-    const clients = webSocketInitializer.roomClients.get(roomId) || [];
-    const clientws = webSocketInitializer.userClients.get(user);
-    clients.push(clientws);
-
-    for (let i = 1; i < result.data.newChatRoom.playerCharacters.length; i++) {
-      if (result.data.newChatRoom.playerCharacters[i].player) {
-        const clientws = webSocketInitializer.userClients.get(
-          result.data.newChatRoom.playerCharacters[i].player.toString(),
-        );
-        clients.push(clientws);
+    if (!result.error) {
+      // join-chatroom
+      const user = result.data.newChatRoom.owner.toString();
+      const roomId = result.data.newChatRoom._id.toString();
+      const clients = webSocketInitializer.roomClients.get(roomId) || [];
+      const clientws = webSocketInitializer.userClients.get(user);
+      clients.push(clientws);
+
+      for (let i = 1; i < result.data.newChatRoom.playerCharacters.length; i++) {
```

Comment:
como a parte do meio ali do `for` é _um menor que_ pode ser usado o i=0.

Porém nessa lógica se o length for igual a 1 ele executaria?

Reply:
Foi erro mesmo